### ISSUE TITLE
consider not globally injecting leaflet

### ISSUE THREAD

**Issue by danielroe** [Gugustinette/Nuxt-Leaflet#1]
I wonder if it would be possible to avoid the global injection of leaflet here - maybe with an autoimport instead?

https://github.com/Gugustinette/Nuxt-Leaflet/blob/main/src/runtime/leaflet-runtime.ts#L7-L9

**Comment by Gugustinette** [comment]
You mean auto-importing every Leaflet function ?

I'm understanding something like in the [lodash](https://nuxt.com/modules/lodash) module : https://github.com/cipami/nuxt-lodash/blob/master/src/module.ts

**Comment by Gruce** [comment]
> You mean auto-importing every Leaflet function ?
> 
> I'm understanding something like in the [lodash](https://nuxt.com/modules/lodash) module : https://github.com/cipami/nuxt-lodash/blob/master/src/module.ts

Because when I don't need Leaflet in a page, it would be injected so high size of page and downloading not necessary scripts.

**Comment by luksak** [comment]
Oh, I didn't read the OP by @danielroe.

What do you mean with autoimport? I saw some modules like use [addImports](https://nuxt.com/docs/api/kit/autoimports#addimports) Is this the way to go?

**Comment by Gugustinette** [comment]
> Oh, I didn't read the OP by @danielroe properly.
> 
> What do you mean with autoimport? I saw some modules like use [addImports](https://nuxt.com/docs/api/kit/autoimports#addimports) Is this the way to go?

Yup that should be it, I'll definitely look into that !

**Comment by Gugustinette** [comment]
> @Gugustinette now I understand what you meant in your comment.
> 
> I looked up some examples. There are a lot of approaches. Here are two examples. https://github.com/AlexLavoie42/Nuxt-Mapbox/blob/main/src/module.ts#L68 https://github.com/hypernym-studio/nuxt-anime/blob/main/src/module.ts#L34 https://github.com/fumeapp/dayjs/blob/main/src/module.ts#L112
> 
> I guess i like the one from nuxt-anime the best. It provides a composable that imports anime: https://github.com/hypernym-studio/nuxt-anime/blob/main/src/runtime/composables/use-anime.ts
> 
> Then it does this:
> 
> ```
> const composablesImport = resolve('./runtime/composables')
> ...
> addImports([{ name: 'useAnime', from: composablesImport }])
> ```
> 
> Simple, understandable an probably the best solution here. Opinions?

That should work, I'll try it ! Thanks for the searches !

**Comment by Gugustinette** [comment]
@luksak @danielroe So I tried a few things :

### Window Error
First, everything led me to the "window is not defined" error.
No matter what I do, somehow Leaflet is trying to execute on the server.
I wonder if it is caused by Leaflet's behavior, like these kind of lines :
https://github.com/Leaflet/Leaflet/blob/35e053c6ff550991a0686865b87a34726e565ce5/src/core/Browser.js#L27

### Bundle Size
It seems like the nuxt-anime library is also affected by the fact that anime.js is imported everywhere, even if you don't use it. However, it's less important with anime.js which represent like 40KB, far from the 2.9MB of Leaflet lol

**Comment by Gugustinette** [comment]
Yeah definitely, but there's clearly a problem with Leaflet calling the `window` constant in all context, which seams to causes error in server-side context.

As far as I understood, this is what makes the library impossible to load on demand in frameworks like Nuxt ?

> Leaflet is only for client thats the reason of the window error. [Read more here](https://github.com/Leaflet/Leaflet/issues/8327)

**Comment by zjrwdl** [comment]
> Because when I don't need Leaflet in a page, it would be injected so high size of page and downloading not necessary scripts.

@Gugustinette Is there any solution to this problem now?

**Comment by Gugustinette** [comment]
I still didn't find any solution at the moment 😭

I'd be glad if someone figure this out.

**Comment by Gugustinette** [comment]
Things are going forward here : https://github.com/Leaflet/Leaflet/pull/9317

I guess these changes will make auto-imports working for Leaflet. 👍

**Comment by Gugustinette** [comment]
So I just realized Vue-Leaflet has some code to import Leaflet automatically when a map is initialized : right [here](https://github.com/vue-leaflet/vue-leaflet/blob/master/src/components/LMap.vue#L250-L256).

This means we can literally remove the import from Nuxt Leaflet to make it better lol.

I made a release [here](https://github.com/Gugustinette/Nuxt-Leaflet/releases/tag/1.0.13) and pushed to NPM.
This is a mistake I made, so I'd like to apologize. 🫡

This is possible thanks to the "smart/lazy" import in Vue-Leaflet, but this kind of problem remains in similar libraries.
